### PR TITLE
adding testing to all notebooks

### PR DESCRIPTION
**Pull request by rawkintrevo**
I'm committing these one at a time. pls don't merge

### CODE REVIEW

**Comment by rawkintrevo** [comment]
I need help from who ever wrote this on `code_summarization.ipynb`

error log- https://github.com/ibm-granite-community/granite-code-cookbook/actions/runs/10781371711/job/29899150657

**Comment by rawkintrevo** [comment]
also

recipes/CodeLLM_DevKit/generate_unit_tests.ipynb \

need to make issues for fixing this error and adding these tests back in

**Comment by adampingel** [comment]
Thanks, @rawkintrevo .  Will look tonight.  I'll err on the side of getting this merged asap so that we've got the pattern established on `main`.  I'll capture any feedback/refinements in separate tickets.

File: recipes/Text_to_Shell/Text_to_Shell.ipynb [hunk 1]
```diff
@@ -35,7 +72,7 @@
    "metadata": {},
    "source": [
     "\n",
-    "## One-shot Prompt with Granite Code 20b\n",
+    "## One-shot Prompt with Granite Code 3b\n",
```
Comment (adampingel) [inline recipes/Text_to_Shell/Text_to_Shell.ipynb:75]: I think on these we should just leave the size out wherever we can (in writing).

In the section where we call `find_langchain_model` we can either have some commented line that finds 20b with another note saying this is how to get the best quality.

File: recipes/Text_to_Shell_Exec/Text_to_Shell_Exec.ipynb [hunk 2]
```diff
@@ -162,20 +178,20 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def chat(prompt: str, \n",
+    "def chat(prompt: str,\n",
     "         system_prompt:str = default_system_prompt,\n",
-    "         examples: str = examples, \n",
-    "         model: str ='granite-code:8b') -> str:\n",
+    "         examples: str = examples,\n",
+    "         model: str ='granite-code:3b') -> str:\n",
```
Comment (adampingel) [inline recipes/Text_to_Shell_Exec/Text_to_Shell_Exec.ipynb:184]: Noting the downstep here as well.  An argument for using Replicate as the default model platform would be that we can leave in the large models as defaults.

File: recipes/Getting_Started_with_Granite_Code.ipynb [hunk 3]
```diff
@@ -92,8 +140,10 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "model_id = \"ibm-granite/granite-8b-code-instruct-128k\"\n",
-    "# model_id = \"ibm-granite/granite-20b-code-instruct-8k\""
+    "from ibm_granite_community.langchain_utils import find_langchain_model\n",
+    "\n",
+    "colab_model = find_langchain_model(platform=\"Replicate\", model_id= \"ibm-granite/granite-8b-code-instruct-128k\")\n",
```
Comment (adampingel) [inline recipes/Getting_Started_with_Granite_Code.ipynb:145]: Can we refer to this as the `replicate_model`?

File: recipes/CodeLLM_DevKit/validating_code_translation.ipynb [hunk 4]
```diff
@@ -55,6 +55,60 @@
     "Code translation aims to convert source code from one programming language to another. Given the promising abilities of large language models (LLMs) in code synthesis, researchers are exploring their potential to automate code translation. In our recent work, [presented at ICSE'24](https://dl.acm.org/doi/10.1145/3597503.3639226), we found that LLM-based code translation is very promising. In this example, we will walk through the steps of translating a Java class to Python and checking various properties of translated code (e.g., number of methods, number of fields, formal arguments, etc.) as a simple form of translation validation."
    ]
   },
+  {
```
Comment (adampingel) [inline recipes/CodeLLM_DevKit/validating_code_translation.ipynb:58]: Can we move all of this ollama stuff to `utils`?  Lazily call if a notebook calls `find_langchain_model`

File: recipes/Getting_Started_with_Granite_Code.ipynb [hunk 5]
```diff
@@ -69,12 +72,57 @@
    "outputs": [],
    "source": [
     "import getpass, os\n",
+    "from ibm_granite_community.notebook_utils import is_colab\n",
     "\n",
-    "replicate_api_token = getpass.getpass()\n",
-    "\n",
-    "os.environ[\"REPLICATE_API_TOKEN\"] = replicate_api_token"
+    "if is_colab():\n",
```
Comment (adampingel) [inline recipes/Getting_Started_with_Granite_Code.ipynb:77]: Can we also factor this into `utils` and lazily call if the user invokes a `find_langchain_model` with "replicate" as the model platform?

File: recipes/CodeLLM_DevKit/validating_code_translation.ipynb (hunk 4 above)
Comment (adampingel) [reply recipes/CodeLLM_DevKit/validating_code_translation.ipynb:58]: Another option is to instead

1. factor the check of whether this stuff is installed into `utils`.  `find_langchain_model` could lazily call the check.
2. If the check fails, print out a helpful error message
3. Default to using Replicate for the automated testing, with a block of code at the top of the notebooks making it easy to comment-in Ollama

I think the previous suggestion is preferable, but I'm including this for completeness.

**Review by adampingel — APPROVED** [review]

File: recipes/CodeLLM_DevKit/validating_code_translation.ipynb (hunk 4 above)
Comment (adampingel) [reply recipes/CodeLLM_DevKit/validating_code_translation.ipynb:58]: Yet another option is to move (some of?) that stuff even earlier into the container image.

+ corresponding instructions for local users (or provide a `.devcontainer.json`?)

**Comment by rawkintrevo** [comment]
Just read through your review adam. All good points, I'll make tickets for each

**Pull request merged**